rust/rust.bzl

```diff
@@ -198,7 +198,10 @@ def _find_crate_root_src(srcs, file_names=["lib.rs"]):
       return src
   fail("No %s source file found." % " or ".join(file_names), "srcs")
 
-def _determine_lib_name(name, crate_type, toolchain):
+def _determine_output_hash(lib_rs):
+  return repr(hash(lib_rs.path))
```

Comment:
@mfarrugi Can you take another look at this PR, specifically this line? I again changed the way the uniquifying hash is generated as the available information changed.

Before: hash output rlib name
Now: hash full lib.rs path (since the rlib name _contains_ the hash in it)

Can you see any downsides in this approach or perhaps have a better thing to hash?

Reply:
I can think of the degenerate case where multiple targets use the same lib.rs, so maybe hash(name, lib.rs path) or Label? Though I don't recall what info is available.

Reply:
I don't think the name is useful as a component as a hash collision will only matter if the crate names are already the same...

I don't think changing this function later will be breaking or onerous, so I'll go ahead and get this submitted.